src/components/dialogs/notificationDialogUtils/index.js

```diff
@@ -0,0 +1,17 @@
+import NotificationViewButton from './NotificationViewButton';
+import NotificationCollaborationGrantAccessButton from './NotificationCollaborationGrantAccessButton';
+import NotificationDeclineCollaborationRequestButton from './NotificationDeclineCollaborationRequestButton';
+
+export const notificationButtonTypes = {
+  view: 'view',
+  grant: 'grant',
+  decline: 'decline',
+  revoke: 'revoke',
+  request: 'request',
+};
+
+export const notificationButtons = {
+  [notificationButtonTypes.view]: NotificationViewButton,
+  [notificationButtonTypes.grant]: NotificationCollaborationGrantAccessButton,
+  [notificationButtonTypes.decline]: NotificationDeclineCollaborationRequestButton,
```

Comment:
TBH i'm still not totally sold on this approach of treating the buttons separately from the notification content. i think the question of "where do we put the error that happens in the button component" is a symptom of having separated two things that can't be easily treated separately.

Reply:
As in, you're envisioning the components in e.g., NotificationViewButton, etc. to be in the notificationSchema file?

Reply:
In my vision each notification has its own file, content and buttons together. But if you are sure this is working I'm ok with that. It just confuses me a little

Reply:
It is working. And I agree; feels a little overkill-y with the abstraction.